### PR TITLE
🌱 Fix lint issues: govet linter

### PR DESCRIPTION
**Pull request by chrismcgehee**
* **Please check if the PR fulfills these requirements**
- [ n/a] Tests for the changes have been added (for bug fixes / features)
- [x] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Code quality improvement

* **What is the current behavior?** (You can also link to an open issue here)
This PR fixes issues from the [govet](https://golang.org/cmd/vet/) linter. With current settings, the only checks enabled are for field alignment and invalid struct field tags. The `fieldalignment` check will look for structs that could take up less memory if the fields are re-ordered. It can also give the optimal ordering which is what I used for this commit. \
Part of issue [#173](https://github.com/ossf/scorecard/issues/173).

* **What is the new behavior (if this is a feature change)?**
n/a

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No. With my latest commit, there is no breaking change in this PR.

### CODE REVIEW

**Comment by naveensrinivasan** [comment]
Thank you! for taking all the warnings. 

IMO For now let’s not have any breaking changes on the JSON. 

I will let @azeemshaikh38 @inferno-chromium and @oliverchang  also chime in.

**Review by azeemshaikh38 — CHANGES_REQUESTED** [review]
There is a chance this will break our BQ data transfer service, since AFAICT, jagged rows are only allowed in CSV format (
image.

I would err on the side of caution and not risk breaking it for now. We can revisit this later.

File: checker/check_result.go [hunk 1]
```diff
@@ -17,12 +17,12 @@ package checker
 const MaxResultConfidence = 10
 
 type CheckResult struct {
+	Error       error `json:"-"`
 	Name        string
-	Pass        bool
+	Details     []string `json:",omitempty"`
```
Comment (azeemshaikh38) [inline checker/check_result.go:22]: Let's remove the "omitempty" option here. That should allow us to submit the PR without any breaking change?

Comment (chrismcgehee) [reply checker/check_result.go]: Yep, that will make the PR not have any breaking changes. I have removed that option.

**Review by azeemshaikh38 — APPROVED** [review]

**Comment by chrismcgehee** [comment]
This one's ready to merge.

**Pull request merged**